### PR TITLE
feat: show all teams within domain

### PR DESCRIPTION
**Pull request by nickoferrall**
Fix: https://github.com/ParabolInc/parabol/issues/8483

@mattkrick we discussed this PR a while ago, and you mentioned privacy concerns, specifically around showing the Team Lead email. I spoke with the Squad, and the sales team currently send a Google Sheet with this info by email. That doesn't make it OK, but it should make it more secure than what we're currently doing. 

I added a comment to the `canViewTeamsInDomain` feature flag to warn future devs to be careful with this flag, and, if this PR is approved, I'll include a warning in the Notion doc too. 

Demo: https://www.loom.com/share/edaf3ff794714b249b17a3835ef55623

If this PR is approved, I'll create a Notion doc with an example mutation and let the sales team know. 

### To test

- [ ] Add the `canViewTeamsInDomain` and `checkoutFlow` user feature flags
- [ ] Go to the Teams view of an org that has the same `activeDomain` as the viewer's domain, e.g. `Parabol Org` has the `parabol.co` domain, and your email is `parabol.co`
- [ ] See that all teams within the domain are visible
- [ ] You can filter to see all teams within org
- [ ] Go to an org with an active domain that's different to the viewer's domain, and you don't see the filter
- [ ] Users without the `canViewTeamsInDomain` can only see teams within their org and don't see the filter menu

### CODE REVIEW

File: packages/client/modules/userDashboard/components/OrgTeams/OrgTeams.tsx [hunk 1]
```diff
@@ -1,49 +1,118 @@
...
+    queryRef
   )
-  const {teams, isBillingLeader} = organization
+
+  const {togglePortal, originRef, menuPortal, menuProps} = useMenu(MenuPosition.UPPER_RIGHT)
+  const {viewer} = data
+  const {featureFlags, domains, organization} = viewer
+  const {activeDomain, teams, isBillingLeader, name} = organization ?? {}
+
+  const ALL_TEAMS_IN_ORG = `All Teams In ${name}`
```
Comment (nickoferrall) [inline packages/client/modules/userDashboard/components/OrgTeams/OrgTeams.tsx:63]: Originally, I added these labels to `FilterLabels` in `constEnums`, but I think we should show the org/domain name, so I'm instantiating them here

File: packages/client/modules/userDashboard/components/OrgTeams/OrgTeams.tsx [hunk 2]
```diff
@@ -1,49 +1,118 @@
...
   return (
     <>
       <h1>{'Teams'}</h1>
+      {canViewTeamsInDomain && (
+        <>
+          <DashFilterToggle onClick={togglePortal} label={label} ref={originRef} value={label} />
+          {menuPortal(
+            <Menu
+              keepParentFocus
+              defaultActiveIdx={label === ALL_TEAMS_IN_DOMAIN ? 0 : 1}
```
Comment (nickoferrall) [inline packages/client/modules/userDashboard/components/OrgTeams/OrgTeams.tsx:92]: If we turn on the feature flag, I assume we'll be directing users to the teams within the domain, so I chose this as the default active index

**Comment by nickoferrall** [comment]
Hey @acressall, it'd be great to get your eyes on the design 🙏  I went with a filter menu instead the switch button I suggested in the issue because I think it's more consistent with our design, and we already have a dropdown menu built

**Review by acressall — APPROVED** [review]
Filter is a good choice!

File: packages/client/modules/userDashboard/components/OrgTeams/OrgTeams.tsx [hunk 3]
```diff
@@ -1,49 +1,118 @@
...
+
+  const {canViewTeamsInDomain} = featureFlags
+
+  const handleMenuItemClick = (newLabel: string) => {
+    setLabel(newLabel)
+    togglePortal()
+  }
+
+  const selectedTeams =
+    (label === ALL_TEAMS_IN_DOMAIN && canViewTeamsInDomain ? teamsByDomain : teams) ?? []
```
Comment (igorlesnenko) [inline packages/client/modules/userDashboard/components/OrgTeams/OrgTeams.tsx:81]: There is a rare edge case. If rename org to `parabol.co`, toggle breaks so I can't switch it.

**Comment by nickoferrall** [comment]
Updated! I was able to see the teams when you couldn't because I had `insights` turned on which meant `suggestedTier` was true for me: https://github.com/ParabolInc/parabol/blob/74b7e46b72637a0a3cc7c9c5c349b1f7e0249a49/packages/server/graphql/public/types/User.ts#L62

I'm now querying the data from the `company` field rather than `domains`, so you should be able to see it now 👍

**Review by igorlesnenko — CHANGES_REQUESTED** [review]
I have a team that in example.com domain

But it shows a teams from parabol.co domain

image

image

**Review by igorlesnenko — CHANGES_REQUESTED** [review]
If I now go to my example.com org, I don't see any filter anymore

image

**Review by igorlesnenko — APPROVED** [review]
As far as I understand if my own domain is different from org domain, I can't see the filter now

**Comment by nickoferrall** [comment]
> As far as I understand if my own domain is different from org domain, I can't see the filter now

Yep, exactly. Your domain is `parabol.co` so `company.organizations` will only return the orgs in your `parabol.co` domain (see [here](https://github.com/ParabolInc/parabol/blob/0f638bcce554c7adea5ac07043d3d7f373a0250b/packages/server/graphql/public/types/Company.ts#L21)). 

If you have the `canViewTeamsInDomain` feature flag, you can go to `Parabol Org` and see all the teams in the org & the domain.

If you go to `Example Org` created by someone with the `example.com` domain, you shouldn't be able to view all the teams in that domain, so I'm hiding the filter.

File: packages/server/graphql/public/types/Company.ts [hunk 4]
```diff
@@ -20,6 +21,8 @@ const getSuggestedTierOrganizations = async (
   const organizations = await dataLoader.get('organizationsByActiveDomain').load(domain)
   const orgIds = organizations.map(({id}) => id)
   const viewerId = getUserId(authToken)
+  const viewer = await getUserById(viewerId)
```
Comment (mattkrick) [inline packages/server/graphql/public/types/Company.ts:24]: +1 how can i test this? to fetch `{viewer { company { id } }` we already have to get the viewer, so I'm not sure why it's needed? if it is needed, could we use the dataloader to reduce a fetch?

**Pull request closed without merging**